File: 2377-check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.java
public class Solution {
    public boolean digitCount(String num) {
        int n = num.length();
        int[] counts = new int[n]; // Initialize counts array with size n
        
        for (int i = 0; i < n; ++i) {
            int digit = num.charAt(i) - '0'; // Convert character to integer value
            if (digit >= n) {
                return false; // If the digit is out of range, return false
            }
            counts[digit]++; // Increment count for the digit at its index
        }
        
        for (int i = 0; i < n; ++i) {
            if (counts[i] != (num.charAt(i) - '0')) {
                return false; // If count doesn't match the digit at its index, return false
            }
        }
        
        return true; // All conditions hold true
    }
}

File: 2377-check-if-number-has-equal-digit-count-and-digit-value/check-if-number-has-equal-digit-count-and-digit-value.cpp
class Solution {
public:
    bool digitCount(std::string num) {
        int n = num.length();
        vector<int> counts(n, 0);         
        for (int i = 0; i < n; ++i) {
            int digit = num[i] - '0'; 
            if (digit >= n) {
                return false; 
            }
            counts[digit]++; 
        }        
        for (int i = 0; i < n; ++i) {
            if (counts[i] != (num[i] - '0')) {
                return false; 
            }
        }        
        return true; 
    }
};
